pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -193,14 +193,32 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step,
 			})
 		}
 
-		// add steps for subscriptions for bundles that were added through resolution
+		// add steps for subscriptions for bundles that were added through resolution and bundles that have been marked deprecated
 		for sub := range existingSubscriptions {
-			if sub.Status.CurrentCSV == op.Name {
-				continue
+			updated := false
+			if len(op.Deprecations) == 0 && corev1.ConditionStatus(sub.Status.GetCondition(v1alpha1.SubscriptionOperatorDeprecated).Type) != corev1.ConditionUnknown {
+				// If new bundle is no longer deprecated then remove the condition
+				sub.Status.RemoveConditions(v1alpha1.SubscriptionOperatorDeprecated)
+				updated = true
+			} else if len(op.Deprecations) > 0 {
+				// Remove old conditions first in case of update
+				sub.Status.RemoveConditions(v1alpha1.SubscriptionOperatorDeprecated)
+				for _, deprecation := range op.Deprecations {
+					sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+						Type:    v1alpha1.SubscriptionOperatorDeprecated,
+						Message: deprecation.Message,
+					})
+				}
```

Comment:
It looks like the result of this will be that the resulting subscription conditions will contain just the last deprecation from `op.Deprecations`. If `type` already exists in the conditions slice `SetCondition` overwrites it.

Reply:
Yes, this part is in a messy in-between state because it's behaving as though we're going to add each deprecation with a different type (package, channel, bundle) but the separate types aren't in there yet so it just overwrites. I'll go ahead and create another `api` PR for the additional condition types then I can clean this up.